### PR TITLE
BugFix: Download upgraded image and add variables to M3M template.

### PR DESCRIPTION
**Pull request by smoshiur1237**
There is a bug in upgrade workflow where upgraded version of image needs to be downloaded from artifactory and pass the upgraded variables in Metal3MachineTemplate.

### CODE REVIEW

File: vm-setup/roles/v1aX_integration_test/tasks/upgrade.yml [hunk 1]
```diff
@@ -441,6 +450,9 @@
       CLUSTER_UID: "{{ clusters.resources[0].metadata.uid }}"
       M3MT_NAME: "{{CLUSTER_NAME}}-new-controlplane-image"
       DATA_TEMPLATE_NAME: "{{CLUSTER_NAME}}-controlplane-template"
+      RAW_IMAGE_NAME: "UBUNTU_20.04_NODE_IMAGE_K8S_{{UPGRADED_K8S_VERSION}}-raw.img"
+      IMAGE_URL: "http://172.22.0.1/images/{{ RAW_IMAGE_NAME }}"
+      IMAGE_CHECKSUM: "http://172.22.0.1/images/{{ RAW_IMAGE_NAME }}.{{ IMAGE_CHECKSUM_TYPE }}sum"
```
Comment (kashifest) [inline vm-setup/roles/v1aX_integration_test/tasks/upgrade.yml:455]: Shouldn't this have the same set of vars as above?

Comment (smoshiur1237) [reply vm-setup/roles/v1aX_integration_test/tasks/upgrade.yml:455]: Other two variables are not needed in  template file. Those two are necessary during downloading the image.

File: vm-setup/roles/v1aX_integration_test/tasks/download_image.yml [hunk 2]
```diff
@@ -18,7 +18,7 @@
 
           - name: Verify specific centos image containing newer version of cloud-init is downloaded
             get_url:
-              url: "{{ IMAGE_LOCATION_CENTOS }}"
+              url: "{{ IMAGE_LOCATION_CENTOS }}/{{ IMAGE_NAME_CENTOS }}"
```
Comment (kashifest) [inline vm-setup/roles/v1aX_integration_test/tasks/download_image.yml:21]: I think this `IMAGE_LOCATION_CENTOS `was also wrong @smoshiur1237 , I think we should use `IMAGE_LOCATION` and `IMAGE_NAME` only here as well. @fmuyassarov can you confirm?

File: vm-setup/roles/v1aX_integration_test/tasks/download_image.yml [hunk 3]
```diff
@@ -18,7 +18,7 @@
...
               dest: "{{ IRONIC_IMAGE_DIR }}/{{ IMAGE_NAME_CENTOS }}"
```
Comment (kashifest) [inline vm-setup/roles/v1aX_integration_test/tasks/download_image.yml:22]: If you change that @smoshiur1237 , would you also change all the occurrences of  `IMAGE_NAME_CENTOS` like this

**Review by kashifest — COMMENTED** [review]
Actually the whole download_image needs refactoring, I think we don't need separate tasks for centos and ubuntu  and its also limiting download of other images defined in lib/common.sh. I ll open a refactor PR for it. Leave it as it is now @smoshiur1237

File: vm-setup/roles/v1aX_integration_test/tasks/download_image.yml (hunk 3 above)
Comment (smoshiur1237) [reply vm-setup/roles/v1aX_integration_test/tasks/download_image.yml:22]: Yes the download file needs optimization. It can have the same name for both OS.

**Review by kashifest — APPROVED** [review]

**Pull request merged**